addons/rating/models/rating_mixin.py

```diff
@@ -27,9 +27,24 @@ class RatingMixin(models.AbstractModel):
 
     @api.depends('rating_ids.rating', 'rating_ids.consumed')
     def _compute_rating_last_value(self):
+        # Pure SQL instead of calling read_group to allow ordering array_agg
+        if not self.ids:
+            self.rating_last_value = 0
+            return
+        self.env.cr.execute("""
+            SELECT
+                array_agg(rating ORDER BY write_date DESC) AS "ratings",
+                res_id as res_id
+            FROM "rating_rating"
+            WHERE
+                res_model = %s
+            AND res_id in %s
+            AND consumed = true
+            GROUP BY res_id""", [self._name, tuple(self.ids)])
+        read_group_raw = self.env.cr.dictfetchall()
+        rating_by_res_id = {e['res_id']: e['ratings'][0] for e in read_group_raw}
```

Comment:
vfe comment: Isn't this ordering perf problem fixed by the private _read_group added in https://github.com/odoo/odoo/pull/84908 ?
(Not 100% sure, but I thought of this change when reading your above comment and diff)

Reply:
@Feyensv It does not exactly do the same. The issue that @tivisse is solving here is that a record could have several ratings from a customer. Calling `_rating_get_access_token` will create new rating.
We are only interested by the last one. I think the SQL request is better.
Using _read_group like this will not give the expected result:
```py
read_group_res = self.env['rating.rating']._read_group(
            [('res_model', '=', self._name), ('res_id', 'in', self.ids), ('consumed', '=', True)],
            ['rating'],
            ['res_id'],
            orderby='write_date')
rating_by_res_id_me = {r['res_id']: r['rating'] for r in read_group_res}
```

FYI I made my tests with the following code:

```py
    def test_rating(self):
        self.subscription.action_confirm()
        access_token = self.subscription._rating_get_access_token()
        access_bis = self.subscription._rating_get_access_token()
        access_3 = self.subscription._rating_get_access_token()
        self.subscription.flush()
        self.subscription.rating_apply(3, token=access_token, feedback="moyen content")
        self.subscription.rating_apply(5, token=access_bis, feedback="super content")
        self.subscription.rating_apply(1, token=access_3, feedback="pas content du tout")
        self.subscription.with_context(arj=True)._compute_rating_last_value()
```

Reply:
Oh, ok I missed a point indeed.
It might be feasible somehow if you manually specify the aggregators to the `read_group`, but it'd still return the data in a complicated format and with useless data.
Thanks for the explanation :).